snippets/base/templates/base/includes/snippet_js.html

```diff
@@ -117,66 +117,61 @@
         };
     }
 
-    // gSnippetsMap polyfill, available in Firefox 22 and above.
-    var gSnippetsMap = null;
-    if (supportsLocalStorage()) {
-        // localStorage is available, so we wrap it with gSnippetsMap.
-        gSnippetsMap = {
-            set: function(key, value) {
-                localStorage[key] = value;
-            },
-            get: function(key) {
-                return localStorage[key];
-            }
-        };
-        window.gSnippetsMap = gSnippetsMap;
-    } else {
-        // localStorage isn't available, use gSnippetsMap (backed by IndexedDB).
-        gSnippetsMap = window.gSnippetsMap;
-    }
+    var gSnippetsMap = window.gSnippetsMap;
+
+    // By the time show_snippet event reaches #snippets the snippet
+    // will have finished initializing and altering the DOM. It's the
+    // time to modifyLinks() and addSnippetBlockLinks().
+    var snippets = document.getElementById('snippets');
+    snippets.addEventListener('show_snippet', function(event) {
+        var topSection = document.getElementById('topSection');
+        // Add sample rate and snippet ID to currently displayed links.
+        var parameters = ('sample_rate=' + SNIPPET_METRICS_SAMPLE_RATE + '&snippet_name=' +
+                          ABOUTHOME_SHOWN_SNIPPET.id);
+        modifyLinks(topSection.querySelectorAll('a'), parameters);
+        addSnippetBlockLinks(topSection.querySelectorAll('.block-snippet-button'));
+        sendImpression();
+    });
 
+    var snippetContainer;
```

Comment:
Store this element outside the `injectSnippet` scope so `injectSnippet` and `removeSnippet` can share its reference.